extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -212,7 +212,10 @@ void HTTPClient::setContentType(std::string content_type) {
 }
 
 std::string HTTPClient::escape(std::string string_to_escape) {
-  return curl_easy_escape(http_session_, string_to_escape.c_str(), gsl::narrow<int>(string_to_escape.length()));
+  char* escaped_chars = curl_easy_escape(http_session_, string_to_escape.c_str(), gsl::narrow<int>(string_to_escape.length()));
+  std::string escaped_string(escaped_chars);
+  curl_free(escaped_chars);
+  return escaped_string;
```

Comment:
In theory string's constructor can throw which would result in leaking `escaped_chars`. Consider using `unique_ptr` with a custom curl deleter.
```suggestion
  struct curl_deleter { void operator()(void* p) noexcept { curl_free(p); } };
  std::unique_ptr<char, curl_deleter> escaped_chars{curl_easy_escape(http_session_, string_to_escape.c_str(), gsl::narrow<int>(string_to_escape.length()))};
  std::string escaped_string(escaped_chars.get());
  return escaped_string;
```